### PR TITLE
Added e2e ActiveGate tests

### PR DESCRIPTION
**Pull request by mjgrzybek**
# Description

ActiveGate e2e tests using [kuttl](https://kuttl.dev) framework.
Dynakube object is modified on the fly and consecutive tests expect AG-corresponding k8s entities to adjust (pods, statefulset, service).

## How can this be tested?
1. Install [kind](https://kind.sigs.k8s.io/)
2. `export APIURL=...; export APITOKEN=...; export PAASTOKEN=...`
4. `make kuttl-activegate`

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly

### CODE REVIEW

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/20-routing.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: dynatrace.com/v1beta1
+kind: DynaKube
+metadata:
+  name: dynakube
+  namespace: dynatrace
+spec:
+  apiUrl: https://acc27517.dev.dynatracelabs.com/api
```
Comment (chrismuellner) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/20-routing.yaml:7]: this should use the environment variable

**Review by chrismuellner — DISMISSED** [review]

**Comment by 0sewa0** [comment]
Nitpick: The numbering of the steps are a bit all over the place 😅

**Comment by mjgrzybek** [comment]
> Nitpick: The numbering of the steps are a bit all over the place sweat_smile

oldschool :smile:  https://stackoverflow.com/a/2435499

File: src/testing/kuttl/activegate/create-dynakube-base.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (chrismuellner) [inline src/testing/kuttl/activegate/create-dynakube-base.sh:1]: why is the "dynakube base" in the `activegate` folder?

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+          valueFrom:
+            secretKeyRef:
+              key: communication-endpoints
+              name: dynakube-activegate-tenant-secret
+        - name: DT_TENANT
+          valueFrom:
+            secretKeyRef:
+              key: tenant-uuid
+              name: dynakube-activegate-tenant-secret
+      image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml:68]: breaks tests

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  hostname: dynakube-activegate-0
+  imagePullSecrets:
+    - name: dynakube-pull-secret
+  initContainers:
+    - args:
+        - "-c"
+        - "/opt/dynatrace/gateway/k8scrt2jks.sh"
+      command:
+        - "/bin/bash"
+      image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml:105]: same

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+              valueFrom:
+                secretKeyRef:
+                  key: communication-endpoints
+                  name: dynakube-activegate-tenant-secret
+            - name: DT_TENANT
+              valueFrom:
+                secretKeyRef:
+                  key: tenant-uuid
+                  name: dynakube-activegate-tenant-secret
+          image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml:179]: same

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+      dnsPolicy: ClusterFirst
+      imagePullSecrets:
+        - name: dynakube-pull-secret
+      initContainers:
+        - args:
+            - "-c"
+            - "/opt/dynatrace/gateway/k8scrt2jks.sh"
+          command:
+            - "/bin/bash"
+          image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/35-assert.yaml:214]: same

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/45-assert.yaml [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+              key: communication-endpoints
+              name: dynakube-activegate-tenant-secret
+        - name: DT_TENANT
+          valueFrom:
+            secretKeyRef:
+              key: tenant-uuid
+              name: dynakube-activegate-tenant-secret
+        - name: DT_DNS_ENTRY_POINT
+          value: https://$(DYNAKUBE_ACTIVEGATE_SERVICE_HOST):$(DYNAKUBE_ACTIVEGATE_SERVICE_PORT)/communication
+      image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/45-assert.yaml:70]: same

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/45-assert.yaml [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+                  key: communication-endpoints
+                  name: dynakube-activegate-tenant-secret
+            - name: DT_TENANT
+              valueFrom:
+                secretKeyRef:
+                  key: tenant-uuid
+                  name: dynakube-activegate-tenant-secret
+            - name: DT_DNS_ENTRY_POINT
+              value: https://$(DYNAKUBE_ACTIVEGATE_SERVICE_HOST):$(DYNAKUBE_ACTIVEGATE_SERVICE_PORT)/communication
+          image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/45-assert.yaml:177]: same

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/55-assert.yaml [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+              key: communication-endpoints
+              name: dynakube-activegate-tenant-secret
+        - name: DT_TENANT
+          valueFrom:
+            secretKeyRef:
+              key: tenant-uuid
+              name: dynakube-activegate-tenant-secret
+        - name: DT_DNS_ENTRY_POINT
+          value: https://$(DYNAKUBE_ACTIVEGATE_SERVICE_HOST):$(DYNAKUBE_ACTIVEGATE_SERVICE_PORT)/communication
+      image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/55-assert.yaml:70]: same

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/65-assert.yaml [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+                  key: communication-endpoints
+                  name: dynakube-activegate-tenant-secret
+            - name: DT_TENANT
+              valueFrom:
+                secretKeyRef:
+                  key: tenant-uuid
+                  name: dynakube-activegate-tenant-secret
+            - name: DT_DNS_ENTRY_POINT
+              value: https://$(DYNAKUBE_ACTIVEGATE_SERVICE_HOST):$(DYNAKUBE_ACTIVEGATE_SERVICE_PORT)/communication
+          image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/65-assert.yaml:84]: same

File: src/testing/kuttl/activegate/tests/capabilities-walkthrough/65-assert.yaml [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+      dnsPolicy: ClusterFirst
+      imagePullSecrets:
+        - name: dynakube-pull-secret
+      initContainers:
+        - args:
+            - "-c"
+            - "/opt/dynatrace/gateway/k8scrt2jks.sh"
+          command:
+            - "/bin/bash"
+          image: acc27517.dev.dynatracelabs.com/linux/activegate:latest
```
Comment (0sewa0) [inline src/testing/kuttl/activegate/tests/capabilities-walkthrough/65-assert.yaml:126]: same

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**